evap/staff/templates/staff_course_comments_quick.html

```diff
@@ -0,0 +1,368 @@
+                    74: "[data-action=publish]",      // j
+                    75: "[data-action=make_private]", // k
+                    76: "[data-action=hide]",         // l
+                     8: ":not(.btn-outline-secondary)[data-action]", // backspace
+                    13: "[data-url=next-course]",     // enter
+                    77: "[data-repeat=unreviewed]",   // m
+                    78: "[data-repeat=all]"           // n
+                };
+
+                if(actions[event.which]) {
+                    let element = slider.find(actions[event.which]);
+                    // review hotkeys have no toggle behaviour
+                    if(actions[event.which].startsWith("[data-action")) {
+                        element.click();
+                    } else if(element.length) {
+                        element[0].click();
+                    }
+                    event.preventDefault();
+                }
+            });
+
+            let touchX = null;
+
+            slider.on("touchstart", event => {
+                touchX = event.originalEvent.touches[0].pageX;
+            });
+            slider.on("touchmove", event => {
+                if(!touchX) {
+                    return;
+                }
+
+                let active = findActive(2);
+                let width = active.outerWidth();
+                let x = event.originalEvent.touches[0].pageX - touchX;
+                let trigger = width / 10;
+
+                if(Math.abs(x) >= trigger) {
+                    resetTouchSliding();
+                    slide(index - Math.sign(x));
+                    updateButtons();
+                    resetTouchSliding();
+                } else {
+                    active.css({
+                        transform: "translateX(" + x + "px)",
+                        opacity: 1 - Math.abs(x / trigger),
+                        transition: "none"
+                    });
+                    active.position();
+                    active.css("transition", "");
+                }
+            });
+            slider.on("touchend", resetTouchSliding);
+            slider.on("touchcancel", resetTouchSliding);
+
+            function resetTouchSliding() {
+                touchX = null;
+                findActive(2).css({transform: "", opacity: ""});
+            }
+
+            repeat("unreviewed");
+        });
+
+        function slide(to) {
+            to = Math.min(Math.max(to, 0), items.length);
+            let direction = to >= index? "right" : "left";
+            index = to;
+
+            if(index === items.length) {
+                let reviewed = items.filter(":not([data-review])").length === 0;
+                let alert = slider.find(".alert");
+                alert.toggleClass("alert-warning", !reviewed);
+                alert.toggleClass("alert-success", reviewed && items.length !== 0);
+                alert.toggleClass("alert-secondary", reviewed && items.length === 0);
```

Comment:
to me it looks like these three classes correspond to the three spans in the alert. is there a way to put these classes directly into/around these spans?

Reply:
No, they change the color of the alert. As I pointed out in another comment some lines below this (which is still relevant), custom classes could be used to toggle the spans (or misuse the `.alert-*` classes to have some visibility logic for their children). Also the `.alert-secondary` case is a bit of convenience to have not the template decide whether to apply it. It felt strange in both locations, so it ended up here.